### PR TITLE
Web3: Only show onboarding when wallet is needed

### PR DESCRIPTION
**Pull request by Marik-D**
# Summary
A description of the changes proposed in the pull request.

Fixes #number

## Checklist
- [x] Change is small and easy to review (please split big changes into multiple PRs)
- [x] Change is consistent with architecture
- [x] Change is consistent with test architecture
- [x] Change is consistent with naming conventions
- [ ] New code is covered with tests
- [ ] Tests related to old code are updated
- [ ] Documentation is up to date with changes

### CODE REVIEW

File: universal-login-example/src/App.tsx [hunk 1]
```diff
@@ -2,7 +2,8 @@ import React from 'react';
 import Web3 from 'web3';
 import {ULWeb3Provider} from '@universal-login/web3';
 
-const web3 = new Web3(new ULWeb3Provider(new Web3.providers.HttpProvider('https://rinkeby.infura.io')));
+const universalLogin = new ULWeb3Provider(new Web3.providers.HttpProvider('https://rinkeby.infura.io'));
```
Comment (marekkirejczyk) [inline universal-login-example/src/App.tsx]: We should probably parametrize that with either env or sth else.
By default, it should work on dev environment. e.g.

```ts
const httpProviderUrl = ENV.HTTP_PROVIDER_URL || 'https://rinkeby.infura.io';
const universalLogin = new ULWeb3Provider(new Web3.providers.HttpProvider(httpProviderUrl));
```
Alternatively:
```ts
const httpProviderUrlFromUrl = new URL(window.location.href).searchParams.get("httpProviderUrl");
const httpProviderUrl = httpProviderUrlFromUrl || 'https://rinkeby.infura.io';
```

File: universal-login-example/src/App.tsx [hunk 2]
```diff
@@ -2,7 +2,8 @@ import React from 'react';
...
+const web3 = new Web3(universalLogin);
 
 web3.eth.getAccounts()
   .then(accounts => console.log('accounts', accounts))
```
Comment (marekkirejczyk) [inline universal-login-example/src/App.tsx]: Do we still need it?

File: universal-login-sdk/package.json [hunk 3]
```diff
@@ -13,7 +13,7 @@
     "fbemitter": "^2.1.1",
     "lodash.clonedeep": "^4.5.0",
     "openzeppelin-solidity": "^2.3.0",
-    "reactive-properties": "^0.1.0"
+    "reactive-properties": "^0.1.3"
```
Comment (marekkirejczyk) [inline universal-login-sdk/package.json]: Remember to first commit/push, then publish. Now current version in [repo](https://github.com/UniversalLogin/universal-login-website) is 0.1.2

File: universal-login-web3/lib/ui/OnboardingModal.tsx [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+import {Onboarding} from '@universal-login/react';
+import React from 'react';
+
+interface OnboardingModalProps {
+  sdk: any;
+  walletService: any;
+  domains: any;
+}
+
+export function OnboardingModal({sdk, walletService, domains}: OnboardingModalProps) {
```
Comment (marekkirejczyk) [inline universal-login-web3/lib/ui/OnboardingModal.tsx]: We should move it to `universal-login-react` simply as `Onboarding`.
When we do so we should move styles to sass files.
#nextpr

File: universal-login-web3/lib/index.ts [hunk 5]
```diff
@@ -32,6 +33,7 @@ const WALLET = {
 export class ULWeb3Provider implements Provider {
   private sdk: UniversalLoginSDK;
   private walletService: WalletService;
+  private uiController: UIController;
```
Comment (marekkirejczyk) [inline universal-login-web3/lib/index.ts]: I think it is time to extract `ULWeb3Provider` and `JsonRPCRequest`, `JsonRPCResponse`, `Callback<ResultType>` to seperate files.

File: universal-login-web3/lib/index.ts [hunk 6]
```diff
@@ -32,6 +33,7 @@ const WALLET = {
 export class ULWeb3Provider implements Provider {
```
Comment (marekkirejczyk) [inline universal-login-web3/lib/index.ts]: It is time to get rid of `WALLET` and use the one from WalletService.

File: universal-login-example/src/App.tsx (hunk 1 above)
Comment (Marik-D) [reply universal-login-example/src/App.tsx]: Yep, already done in the next PR

File: universal-login-web3/lib/index.ts (hunk 6 above)
Comment (Marik-D) [reply universal-login-web3/lib/index.ts]: Yes, it's in the next PR

File: universal-login-example/src/App.tsx (hunk 2 above)
Comment (Marik-D) [reply universal-login-example/src/App.tsx]: removed

File: universal-login-example/src/App.tsx (hunk 1 above)
Comment (gasiortomasz) [inline universal-login-example/src/App.tsx]: Infura has recently introduced API request limits. Here is our api key
`https://rinkeby.infura.io/v3/b3026fc5137a4bd18e5d5906ed49f77d`

**Pull request merged**